third_party/amd/lib/TritonAMDGPUToLLVM/SchedInstructions.cpp

```diff
@@ -211,32 +286,41 @@ struct InstructionSchedHintsRewriter
     if (numBufferLoadInstB == 0)
       schedHint.emitError("buffer load count for tile B must be initialized");
 
-    const uint32_t numMfmaInst = schedHint.getNumMMAs().getValue();
+    const uint32_t numMmaInst = schedHint.getNumMMAs().getValue();
 
-    auto mfmaType = cast<RankedTensorType>(schedHint.getNumMMAs().getType());
-    const uint32_t nPerXDL = mfmaType.getShape()[1];
-    const uint32_t mfmaCycle = nPerXDL == 16 ? 16 : 32;
+    auto mmaType = cast<RankedTensorType>(schedHint.getNumMMAs().getType());
+    auto maybeMmaExecCycle = machineDescr->getMmaExecCycle(mmaType.getShape());
+    if (llvm::failed(maybeMmaExecCycle))
+      schedHint.emitError("unknown mma instruction type");
```

Comment:
Should we return here? (Similary for other places in this function.)

Reply:
Done